File: Palindrome.rb
start = Time.now
@@largest = 0
def check_palindrome(number)
	original = number.to_s
	if(original == original.reverse)
		if(number > @@largest)
			@@largest = number
		end
		return true
	end
	return false
end

i = 999
begin
	j = 990
	while j > 900
		j -= 1
		test_number = i * j
		check_palindrome(test_number) unless test_number < @@largest
	end
	i -= 1
end while i > 900#!check_palindrome(test_number)
puts @@largest
puts (Time.now - start)*1000

File: Palindrome.java
class Palindrome{
	private static long start = System.nanoTime();
	private static int largestPalindrome;

	public static void main(String[] args) {
		int testNumber = 0;
		largestPalindrome = 0;
		int i = 999;
		do{
			int j = 990;
			while(j > 999){
				j -= 1;
				testNumber = i * j;
				if(testNumber > largestPalindrome)
					check_palindrome(testNumber);
			}
			i--;
		}while(/*!check_palindrome(testNumber)*/i > 900);
		System.out.println(largestPalindrome);
		System.out.println((System.nanoTime() - start));
	}
	
	private static boolean check_palindrome(int number){
		char[] numberString = Integer.toString(number).toCharArray();
		for(int i = 0; i < 1+numberString.length/2; i++){
			if(numberString[i] != numberString[numberString.length - 1 - i]){
				return false;
			}
		}
		if(number > largestPalindrome)
			largestPalindrome = number;
		return true;
	}

}